ethportal-api/src/types/execution/block_body.rs

```diff
@@ -72,6 +72,25 @@ impl ssz::Decode for BlockBody {
     }
 }
 
+/// Calculate the Merkle Patricia Trie root hash from a list of items
+pub fn calculate_root<'a, T: Encodable + 'a>(
+    items: impl Iterator<Item = &'a T>,
```

Comment:
I want to avoid the clone `IntoIterator`

Reply:
We can reassess this in the future if `Iterator` doesn't work for all cases, but currently `Iterator` does work for all of our usecases

Reply:
You don't need to clone, and it is a simple change. Here is how to do it:

This function would change to:
```rust
pub fn calculate_merkle_patricia_root<'a, T: Encodable + 'a>(
    items: impl IntoIterator<Item = &'a T>,
) -> anyhow::Result<B256> {
    let memdb = Arc::new(MemoryDB::new(true));
    let mut trie = EthTrie::new(memdb);

    // Insert items into merkle patricia trie
    for (index, tx) in items.into_iter().enumerate() {
        let path = alloy_rlp::encode(index);
        let encoded_tx = alloy_rlp::encode(tx);
        trie.insert(&path, &encoded_tx)
            .map_err(|err| anyhow!("Error inserting into merkle patricia trie: {err:?}"))?;
    }

    trie.root_hash()
        .map_err(|err| anyhow!("Error calculating merkle patricia trie root: {err:?}"))
}
```
And in other places (e.g. ethportal-api/src/types/execution/block_body.rs), you would have:
```rust
calculate_merkle_patricia_root(&self.transactions()?)
```
instead of 
```rust
calculate_merkle_patricia_root(self.transactions()?.iter())
```
In fact, both syntax would work, but I find second one cleaner.

---

Also in trin-execution/src/era/beacon.rs, you would do `calculate_merkle_patricia_root(&withdrawals)?;`

Reply:
Oh cool I didn't make the link that IntoIterator could be into any type of iterator very cool